src/github.com/stellar/horizon/resource/account.go

```diff
@@ -71,3 +72,16 @@ func (this *Account) Populate(
 	this.Links.Data.PopulateTemplated()
 	return
 }
+
+// GetData returns decoded value for a given key. If the key does
+// not exist, empty slice will be returned.
+func (this *Account) GetData(key string) []byte {
+	data, exists := this.Data[key]
+	if !exists {
+		return nil
+	}
+
+	// We assume that horizon always returns valid base64 encoded string
+	decoded, _ := base64.StdEncoding.DecodeString(data)
```

Comment:
> Unfortunately, putting validation in the ingester won't help. Someone may go in and overwrite a value with SQL.

You mean who? A developer/hacker? If the value is not valid base64 encoded string then it's a bug, right?

> Why can't you handle the errors? Every assumption you make about outside state is just another place for your code to break.

I understand but with error return `GetData` method doesn't add any value. For example, in https://github.com/stellar/go/issues/35 developer wants to access data value bytes directly. So if you compare these two snippets:
```go
bytes, err := account.GetBytes("key")
if err != nil {
    // handle error
}
```
and
```go
bytes, err := base64.StdEncoding.DecodeString(account.Data["key"])
if err != nil {
    // handle error
}
```
you can start thinking that `GetBytes` isn't that helpful since `encoding/base64` is a standard library. Whereas, `GetData(key string) []byte` allows you to do certain things like passing the result to the function directly what's not possible for multiple return functions (ex. `fmt.Println(string(account.GetBytes("key")))`).

My main assuption here is that: _we should trust horizon to have valid, consistent data_. Otherwise, we would have to check every single value returned by it before using it in other apps code including:
* account id: is it valid strkey, is decoded value length correct?
* hash: is the hash valid hex value?
* links: is it valid, does it point to the some host?
* post transaction: is the transaction really in the ledger?
* balance: does this account really have this balance? maybe I should pull all tx affecting this account, check signatures and add up every credits and debits?
* etc.